src/rgw/driver/rados/rgw_notify.cc

```diff
@@ -1087,14 +1106,18 @@ int publish_commit(rgw::sal::Object* obj,
       std::vector<buffer::list> bl_data_vec{std::move(bl)};
       librados::ObjectWriteOperation op;
       cls_2pc_queue_commit(op, bl_data_vec, topic.res_id);
-      const auto ret = rgw_rados_operate(
-	dpp, res.store->getRados()->get_notif_pool_ctx(),
-	queue_name, &op, res.yield);
+      auto *completion = librados::Rados::aio_create_completion();
+      auto *pcc_arg = new PublishCommitCompleteArg{queue_name, dpp};
```

Comment:
probably safer to use use `std::unique_ptr` here as well. you would need to call `release()` if ` ret == 0` after line 1122 (to make sure we don't double delete) and pass the raw pointer to `set_complete_callback()`

Reply:
Right. I thought about that but somehow I thought we could have a race condition around the `pcc_arg` between `publish_commit()` and `publish_commit_completion()`. I see now that that's not an issue as the ret value from `aio_complete()` is a clear indicator of whether we should hold on to the pointer or `release()` it.  I will use unique_ptr for both pcc_arg and completion.